Lisp/tests/syntax_test_lisp.lisp

```diff
@@ -0,0 +1,2865 @@
+;            ^ punctuation.section.parens.begin.lisp
+;             ^^^^^ variable.other.lisp
+;                   ^ punctuation.section.parens.begin.lisp
+;                    ^^^^ meta.function-call.lisp support.function.lisp
+;                         ^^^^^ variable.other.lisp
+;                              ^ punctuation.section.parens.end.lisp
+;                               ^ punctuation.section.parens.end.lisp
+;                                 ^ punctuation.section.parens.begin.lisp
+;                                  ^^^^^ variable.other.lisp
+;                                        ^ punctuation.section.parens.begin.lisp
+;                                         ^^^^ meta.function-call.lisp support.function.lisp
+;                                              ^^^^^ variable.other.lisp
+;                                                   ^^ punctuation.section.parens.end.lisp
+;                                                      ^^ punctuation.section.parens.end.lisp
+
+;########
+; LOOPS #
+;########
+
+(loop
+
+;;; name-clause
+
+  named name
+; ^^^^^ keyword.declaration.name.lisp
+;       ^^^^ entity.name.loop.lisp
+
+;;; initial-final-clause
+
+  initially ()
+; ^^^^^^^^^ keyword.control.flow.lisp
+;           ^^ meta.parens.lisp
+
+  finally (return (values evens odds))
+;^^^^^^^^^ meta.parens.lisp - meta.parens meta.parens
+;         ^^^^^^^^ meta.parens.lisp meta.parens.lisp - meta.parens meta.parens meta.parens
+;                 ^^^^^^^^^^^^^^^^^^^ meta.parens.lisp meta.parens.lisp meta.parens.lisp
+;                                    ^ meta.parens.lisp meta.parens.lisp - meta.parens meta.parens meta.parens
+;                                     ^ meta.parens.lisp - meta.parens meta.parens
+; ^^^^^^^ keyword.control.flow.lisp
+;         ^ punctuation.section.parens.begin.lisp
+;          ^^^^^^ keyword.control.flow.return.lisp
+;                 ^ punctuation.section.parens.begin.lisp
+;                  ^^^^^^ meta.function-call.lisp support.function.lisp
+;                         ^^^^^ variable.other.lisp
+;                               ^^^^ variable.other.lisp
+;                                   ^^ punctuation.section.parens.end.lisp
+
+;;; with-clause
+
+  with var1 :float = 10.5 and var2 = true
+; ^^^^ keyword.declaration.variable.lisp
+;      ^^^^ variable.other.lisp
+;           ^ punctuation.accessor.lisp
+;            ^^^^^ keyword.other.symbol.lisp
+;                  ^ keyword.operator.lisp
+;                    ^^^^ constant.numeric.value.lisp
+;                         ^^^ keyword.operator.logical.lisp
+;                             ^^^^ variable.other.lisp
+;                                  ^ keyword.operator.lisp
+;                                    ^^^^ constant.language.boolean.lisp
+
+;;; for-as-claus
+
+  for x-list = '(a b c) then (cdr x-list)
+; ^^^ keyword.control.loop.lisp
+;     ^^^^^^ variable.other.lisp
+;            ^ keyword.operator.lisp
```

Comment:
Is this strict enough to know if it's an assignment, a comparison, or not?

Reply:
Maybe someone else wants to implement https://www.lispworks.com/documentation/lw70/CLHS/Body/m_loop.htm#loop in a more detailed way, but the rather simple `loob-body` context can't ensure a more detailed meaning, I guess.